docs/resource-sharing/os-backfill-containers.md

```diff
@@ -27,34 +27,146 @@ In order to configure the container, you will need:
-user jobs; this allows the user to utilize a container for their job that is different from the pilot.
+(Replace `/path/to/token` with the location you saved the token obtained from the OSG VO administrators.)
+Privileged mode (`--privileged`) requested in the above `docker run` allows the container
+to mount CVMFS using cvmfsexec (see below) and invoke `singularity` for user jobs.
+Singularity allows the user to use their own container for their job.
+
+
+CVMFS
+-----
+
+CVMFS is a read-only remote filesystem that many OSG jobs depend on for software and data.
+Supporting CVMFS inside your container will greatly increase the types of OSG jobs you can run.
+
+There are two methods for adding CVMFS: [enabling cvmfsexec](#adding-cvmfs-using-cvmfsexec),
+or [bind-mounting CVMFS from the host](#adding-cvmfs-via-bind-mount).
+Bind-mounting CVMFS will require CVMFS to be installed on the host first,
+but the container will need somewhat fewer privileges.
+
+
+### Adding CVMFS using cvmfsexec
+
+[cvmfsexec](https://github.com/CVMFS/cvmfsexec#readme) is a tool that can be used to mount CVMFS inside the container
+without requiring CVMFS to be installed on the host.
+
+To enable cvmfsexec, specify a space-separated list of repos in the `CVMFSEXEC_REPOS` environment variable.
+We recommend the following repos:
+-   `oasis.opensciencegrid.org`
+-   `singularity.opensciencegrid.org`
+
+cvmfsexec has the following system requirements:
+
+-   On EL7, you must have kernel version >= 3.10.0-1127 (run `uname -vr` to check), and user namespaces enabled.
+    See step 1 in the
+    [Singularity Install document](https://opensciencegrid.org/docs/worker-node/install-singularity/#enabling-unprivileged-singularity)
+    for details.
+
+-   On EL8, you must have kernel version >= 4.18 (run `uname -vr` to check).
+
+See the [cvmfsexec README](https://github.com/cvmfs/cvmfsexec#readme) details.
+
+Note that cvmfsexec will not be run if CVMFS repos are already available in `/cvmfs` via bind-mount,
+regardless of the value of `CVMFSEXEC_REPOS`.
+
+Using cvmfsexec takes place in the entrypoint, which means it will still happen
+even if you specify a different command to run, such as `bash`.
+You can bypass the entrypoint by passing `--entrypoint <cmd>` where `<cmd>` is some different command to run,
+e.g. `--entrypoint bash`.
+Setting the entrypoint this way clears the command.
+
+
+#### Additional cvmfsexec configuration
+
+There are several environment variables you can set for cvmfsexec:
+
+-   `CVMFSEXEC_REPOS` - this is a space-separated list of CVMFS repos to mount.
+    Leave this blank to disable cvmfsexec.
+    OSG jobs frequently use the OASIS repo (`oasis.opensciencegrid.org`) and
+    the singularity repo (`singularity.opensciencegrid.org`).
+
+-   `CVMFS_HTTP_PROXY` - this sets the proxy to use for CVMFS; if left blank
+    it will find the best one via WLCG Web Proxy Auto Discovery.
```

Comment:
Not an issue with the documentation but would it be possible to set the CVMFS proxy location in this order?

1. `CVMFS_HTTP_PROXY`
2. `OSG_SQUID_LOCATION`
3. Auto discovery

Reply:
I don't know the answer to that. Perhaps @DrDaveD does.

Reply:
The container could do it.  To use auto discovery set `CVMFS_HTTP_PROXY="auto;DIRECT"` because cvmfsexec sets `CVMFS_PAC_URLS` to the appropriate servers.

Reply:
https://opensciencegrid.atlassian.net/browse/SOFTWARE-4691